### ISSUE TITLE
[Feature] Stitch recordings together

### ISSUE THREAD

**Issue by XilefTech** [ReplayMod/ReplayMod#554]
I would love to see a feature where you can add multiple replay mod recordings into one longer one. 

I see this used for a lot of people that are working on buildings (let's say on a server) and are going offline regularly or their client just crashes every few hours. This leaves them with many small files yet if they want to make one big timelapse shot, they would need to render many different scenes and need to know how to move keyframes from one recording to the next.

Also, this could be useful if you want to send your replay to a friend for example. That way you won't have to send a thousand small files. 

From what I've read on the internet about this topic, there was a feature like that once but it has been removed because there wasn't enough testing? I wonder if that is true and if so, why it has been removed because I see this as a very helpful feature for some users.
Also I was wondering if it was possible to use an older version of replaymod, to stitch my 1.16 replayfiles together without loosing data. That would be at least one way for me to get around this issue for now.

Thanks in advance for a response!

**Comment by Johni0702** [comment]
> From what I've read on the internet about this topic, there was a feature like that once but it has been removed because there wasn't enough testing? I wonder if that is true and if so, why it has been removed because I see this as a very helpful feature for some users.

That's correct. This used to be a feature but it wasn't really ever working well and once upon a item when the GUI for cutting replays was re-written, it was dropped.

> Also I was wondering if it was possible to use an older version of replaymod, to stitch my 1.16 replayfiles together without loosing data. That would be at least one way for me to get around this issue for now.

No, cutting, stitching and most other replay file manipulation is highly dependent on the exact Minecraft version of the replay.

**Comment by XilefTech** [comment]
so how likely is it that this feature will be implemented in the near future and within a 1.16 replaymod release? - would be cool to see it for 1.12 as well since it is still a frequently-used version for modding and some huge projects like BuildTheEarth relie on it.

**Comment by Sebast04** [comment]
I would love that one, but not only with that one, I'll give an example: I regularly play on a pvp Server with other players of which some also use the replaymod, but its hard to make a good video out of it because of renderdistance. i would love if the mod notices that i have multiple replay recordings for a time period (overlaping) with matching minecraft versions that it would ask if you want to combine them in a third one. of course stuff like lag could mess things up, maybe if one client experienced more lag then others you could prioritise the one with less lag when they meet up. would love this but I can see where this is hard to implement

**Comment by XilefTech** [comment]
> I would love that one, but not only with that one, I'll give an example: I regularly play on a pvp Server with other players of which some also use the replaymod, but its hard to make a good video out of it because of renderdistance. i would love if the mod notices that i have multiple replay recordings for a time period (overlaping) with matching minecraft versions that it would ask if you want to combine them in a third one. of course stuff like lag could mess things up, maybe if one client experienced more lag then others you could prioritise the one with less lag when they meet up. would love this but I can see where this is hard to implement

Well, I don't think that this will be seen anytime soon. this seems just way too complicated and niche to be worth the tons of work that (probably) need to go into that. Also, I think that this isn't quite the main topic of this issue.

**Comment by RaderRMT** [comment]
best thing for now would be to wait for me to make a version of [Bob](https://github.com/RaderRMT/Bob) that merges replays

**Comment by XilefTech** [comment]
Bob seems like a really useful and cool project!
Haven't heard about it before but I'll definitely look into it. How long do you think it will take to push an update with the given functionality? (Don't want to seem impatient, just so ik how long I'll have to wait and wether it is worth waiting a few days with my current recording/timelapse)

**Comment by RaderRMT** [comment]
> How long do you think it will take to push an update with the given functionality?

a lot of time. the thing holding me back from making progress is making the user interfaces

> Don't want to seem impatient, just so ik how long I'll have to wait and wether it is worth waiting a few days with my current recording/timelapse

it's currently not worth the wait (you'll have to wait at least a month (if i'm productive) for being able to edit the recording without a gui, and far more if you want to edit your replay with a gui). making bob needs some planning and i'm also re-writing everything. as i said previously, user interfaces are what's holding me back from making progress
